lib/kubeclient/gcp_auth_provider.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'kubeclient/google_application_default_credentials'
+require 'kubeclient/gcp_command_credentials'
+
+module Kubeclient
+  # Handle different ways to get a bearer token for Google Cloud Platform.
+  class GCPAuthProvider
+    class << self
+      def token(config)
+        if config.key?('cmd-path')
+          Kubeclient::GCPCommandCredentials.token(config)
+        else
+          Kubeclient::GoogleApplicationDefaultCredentials.token
```

Comment:
In the interests of hardening, would you be opposed to adding retries here? We have begun observing intermittent `Signet::RemoteServerError` exceptions that, [according to Google](https://cloud.google.com/apis/design/errors), we can/should retry on.

I don't think pushing up retryable errors to the caller is the right thing to do; we should handle them here and only throw once we've exhausted our retry limit

Reply:
@timothysmith0609 sounds like something that the `googleauth` gem should be dealing with, not `kubeclient` itself.

Reply:
You're absolutely right. It turns out `googleauth` does retry, but only when using the GCE node service account (no idea why that's the case). It looks like it should be possible to wrap the other case in their retry routine so I'll try to solve this upstream.